src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1931,18 +1939,43 @@ private BoundExpression BindNonMethod(SimpleNameSyntax node, Symbol symbol, Bind
                             if (IsBadLocalOrParameterCapture(parameter, parameter.Type, parameter.RefKind))
                             {
                                 isError = true;
-                                Error(diagnostics, parameter.Type.IsRefLikeType ? ErrorCode.ERR_AnonDelegateCantUseRefLike : ErrorCode.ERR_AnonDelegateCantUse, node, parameter.Name);
+
+                                if (parameter.RefKind != RefKind.None)
+                                {
+                                    Error(diagnostics, ErrorCode.ERR_AnonDelegateCantUse, node, parameter.Name);
+                                }
+                                else if (parameter.Type.IsRestrictedType(ignoreSpanLikeTypes: true))
```

Comment:
Would it be simpler to remove this case and just updating `ERR_AnonDelegateCantUseRefLike` (used in case below) to say "restricted" instead of "ref-like"?

Reply:
> Would it be simpler to remove this case and just updating ERR_AnonDelegateCantUseRefLike (used in case below) to say "restricted" instead of "ref-like"?

The logic doesn't look particularly complicated to me and we have an existing error for the situation, which was reported for regular parameters during lambda rewrite. In a sense, I am preserving an existing behavior for regular parameters by using ```ErrorCode.ERR_SpecialByRefInLambda``` here.